src/components/spreadsheet/use-spreadsheet-equipments.ts

```diff
@@ -144,7 +148,63 @@ export const useSpreadsheetEquipments = (
                     setIsFetching(false);
                 });
         }
-    }, [equipment, shouldFetchEquipments, studyUuid, currentNode?.id, dispatch, formatFetchedEquipments]);
+    }, [
+        equipment,
+        shouldFetchEquipments,
+        studyUuid,
+        currentNode?.id,
+        dispatch,
+        formatFetchedEquipments,
+        customColumnsDefinitions,
+    ]);
+
+    useEffect(() => {
+        if (studyUuid && equipments != null) {
+            const aliases = customColumnsNodesAliases.map((nodeAlias) => nodeAlias.alias);
+            let fetchers: Promise<unknown>[] = [];
+            let additionalEquipmentsByNodes: Record<string, Record<SpreadsheetEquipmentType, Identifiable[]>> = {};
+            if (equipment.type) {
+                aliases.forEach((alias) => {
+                    const nodeNameAssociatedToAlias = customColumnsNodesAliases.find(
+                        (nodeAlias) => nodeAlias.alias === alias
+                    )?.name;
```

Comment:
directly iterate through customColumnsNodesAliases to get the names ?

Reply:
Can't we store the node UUIDs directly and the names would be only used for displaying purposes in the autocomplete ? But behind the scene we only UUIDs ? I think it would be a bit more robust